desktop/core/src/desktop/js/parse/sql/testUtils.ts

```diff
@@ -0,0 +1,105 @@
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// 'License'); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an 'AS IS' BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import fs from 'fs';
+
+import { AutocompleteParser } from '../types';
+
+interface TestCase {
+  namePrefix: string; // ex. "should suggest keywords"
+  beforeCursor: string;
+  afterCursor: string;
+  containsKeywords?: string[];
+  noErrors?: boolean;
+  expectedResult: {
+    lowerCase?: boolean;
+    suggestTables?: {
+      identifierChain?: { name: string }[];
+      onlyTables?: boolean;
+    };
+    suggestDatabases?: {
+      appendDot?: boolean;
+    };
+  };
+}
+
+interface GroupedTestCases {
+  jisonFile: string;
+  testCases: TestCase[];
+}
+
+/**
+ * Finds and parses x.test.json files given a list of jison files.
+ * For example, if alter_table.jison is part of the structure it will look for alter_table.test.json and if it does
+ * it'll parser it (TestCase[]). Test cases are grouped per found .jison file.
```

Comment:
nit: spelling

Reply:
Ack. I'll updater it in the next set of changes.